### PR TITLE
Fix pages/p/[siteId].js code snippet

### PR DESCRIPTION
**Pull request by matiasleidemer**
Hey there, Lee 👋 

First of all, thank you very much for sharing this amazing course! I'm following it through the documentation, and I found an issue with the `page/p/[siteId].js` code snippet. 

It turns out we should use `useEffect` to set the `allFeedback` state; otherwise, it might be `undefined`, leading to bugs. I'm not sure if this is something you address in the videos, however. I also took the change to update some other little things I noticed. If you want me to update something else, please let me know; I'd be happy to help.

I'm not done with the course yet, so maybe there will be more PRs in the future. 😄

### CODE REVIEW

File: docs/pages/feedback/firebase-admin.md [hunk 1]
```diff
@@ -61,12 +61,16 @@ if (!admin.apps.length) {
...
+</Callout>
+
 ## Retrieving Data
 
-Finally, lets create `lib/db-admin.js` to fetch data from Firestore.
+Finally, lets create `utils/db-admin.js` to fetch data from Firestore.
 This function will fetch all `Feedback` where the `siteId` matches.
 
-**`lib/db-admin.js`**
+**`utils/db-admin.js`**
```
Comment (matiasleidemer) [inline docs/pages/feedback/firebase-admin.md:73]: I understand the final project has those files in `lib/`, however, in the docs, we're using `utils/`, so I decided to update it here for consistency.

File: docs/pages/feedback/feedback-page.md [hunk 2]
```diff
@@ -185,9 +191,13 @@ const FeedbackPage = ({ initialFeedback }) => {
           </FormControl>
         </Box>
       )}
+
       {allFeedback &&
         allFeedback.map((feedback) => (
-          <Feedback key={feedback.id} {...feedback} />
+          <Feedback
+            key={feedback.id || new Date().getTime().toString()}
```
Comment (leerob) [inline docs/pages/feedback/feedback-page.md:198]: Why was this added?

File: docs/pages/feedback/firebase-admin.md (hunk 1 above)
Comment (leerob) [reply docs/pages/feedback/firebase-admin.md:73]: I would think it should be the inverse. All places in the docs should reference `lib` now.

Looking at the file above in this code review, I see:

```
import { getAllFeedback, getAllSites } from '@/lib/db-admin'
```

**Review by leerob — COMMENTED** [review]
Thank you for helping!

File: docs/pages/feedback/feedback-page.md (hunk 2 above)
Comment (matiasleidemer) [reply docs/pages/feedback/feedback-page.md:198]: React complains about the `undefined` key when the feedback was created using the UI (as opposed to loading from Firebase). I wasn't sure this was the best approach given we want to display the new feedback before firebase persists it.

File: docs/pages/feedback/firebase-admin.md (hunk 1 above)
Comment (matiasleidemer) [reply docs/pages/feedback/firebase-admin.md]: This should do the trick https://github.com/leerob/react2025/pull/7/commits/246c918cf84877926ca5bd4743c842b8fa7cea0a

File: docs/pages/feedback/feedback-page.md (hunk 2 above)
Comment (leerob) [reply docs/pages/feedback/feedback-page.md:198]: At a certain point in the course, we change this so that it gets the ID from Firebase of the new document async. I might have just not added it to the docs. Hmm.
